src/libs/ReportUtils.ts

```diff
@@ -1415,6 +1412,23 @@ function canCreateTaskInReport(report: OnyxEntry<Report>): boolean {
     return true;
 }
 
+/**
+ * For all intents and purposes a report that has no notificationPreference at all should be considered "hidden".
+ * We will remove the 'hidden' field entirely once the backend changes for https://github.com/Expensify/Expensify/issues/450891 are done.
+ */
+function isHiddenForCurrentUser(notificationPreference: string | null | undefined): boolean;
+function isHiddenForCurrentUser(report: OnyxEntry<Report>): boolean;
+function isHiddenForCurrentUser(reportOrPreference: OnyxEntry<Report> | string | null | undefined): boolean {
```

Comment:
I think we can avoid overloading here because the number of params for the function and the return type are [same](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads:~:text=Because%20both%20overloads,overloads%20when%20possible). `reportOrPreference` also looks clear enough IMO.

Reply:
Very fair, that would be cleaner